data_structures/src/chain.rs

```diff
@@ -2101,22 +2119,29 @@ impl TransactionsPool {
     pub fn insert(&mut self, transaction: Transaction, fee: u64) -> Vec<Transaction> {
         let key = transaction.hash();
 
+        let mut added_transaction = true;
+        let tx_not_added = transaction.clone();
+
         match transaction {
             Transaction::ValueTransfer(vt_tx) => {
                 let weight = f64::from(vt_tx.weight());
                 let priority = OrderedFloat(fee as f64 / weight);
 
-                self.total_vt_weight += u64::from(vt_tx.weight());
+                if fee < self.minimum_vtt_fee {
+                    added_transaction = false;
```

Comment:
I think you can do an early return here to avoid the transaction clone and the `added_transaction` flag

```suggestion
                    // Do not insert the transaction if the fee is lower than the minimum fee
                    return vec![transaction];
```